plugins/inputs/vsphere/endpoint.go

```diff
@@ -834,6 +844,7 @@ func (e *Endpoint) Collect(ctx context.Context, acc telegraf.Accumulator) error
 	if e.Parent.ObjectDiscoveryInterval.Duration == 0 {
 		err := e.discover(ctx)
 		if err != nil {
+			acc.AddError(err)
```

Comment:
I think this will end up logging the error twice. no need to both log and return it.

Reply:
Will fix this.